scipy/linalg/tests/test_decomp.py

```diff
@@ -901,128 +905,153 @@ def test_deprecation_results(self):
         assert_allclose(v_dep, v)
 
 
-class TestSVD_GESDD:
-    lapack_driver = 'gesdd'
+class TestSVD:
 
     def test_degenerate(self):
         assert_raises(TypeError, svd, [[1.]], lapack_driver=1.)
         assert_raises(ValueError, svd, [[1.]], lapack_driver='foo')
 
-    def test_simple(self):
-        a = [[1, 2, 3], [1, 20, 3], [2, 5, 6]]
+    def check_simple(lapack_driver, xp=None):
```

Comment:
Why changing all the test names? And similarly all the way below

Reply:
This seems to be answered in the PR description:

_TestSVD is a little strange due to the way the lapack_driver parameter is tested. I have tried to apply a minimal refactor here, but a more substantial refactor may result in something more readable. It is a bit of a misnomer that all of our array API compatible tests are under TestSVD_GESDD, just because gesdd is the default value._

It would be nice separating that out into another PR, if there's a refactor here that makes sense regardless of this current PR. And it would make the diff here smaller and easier to understand.

Reply:
Yep, the crux of the issue is that we are trying to only test with other backends for the default value of `lapack_driver`. I'm hoping that there is a smarter refactor which means that our xp tests don't have to be under the `GESDD` label, but I couldn't come up with it myself.